### PR TITLE
Ignore ipynb_checkpoints for flake8 and isort

### PR DESCRIPTION
**Pull request by fehiepsi**
I mainly use jupyterlab for writing scripts/editing files, which leaves many `.ipynb_checkpoints` folders. On the other hand, `flake8` and `isort` act on all files including files in `ipynb_checkpoints` folders. This PR adds ipynb_checkpoints exceptions for these tools.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 [isort]
 line_length = 120
 not_skip = __init__.py
+skip_glob = .ipynb_checkpoints
 known_first_party = pyro, tests
 known_third_party = opt_einsum, six, torch, torchvision
 
 [tool:pytest]
 filterwarnings = error
-    ignore::DeprecationWarning
```
Comment (fehiepsi) [inline setup.cfg]: @neerajprad I think this line is redundant. What do you think?

Comment (fritzo) [reply setup.cfg]: Can someone explain to me why this is redundant?

Comment (fritzo) [reply setup.cfg]: I believe we should instead remove the `once::DeprecationWarning` line. We really do want to ignore `DeprecationWarning`s

Comment (fehiepsi) [reply setup.cfg]: Yeah, let me see what's different when we use `ignore` instead of `once`.

File: setup.cfg [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 line_length = 120
 not_skip = __init__.py
+skip_glob = .ipynb_checkpoints
 known_first_party = pyro, tests
 known_third_party = opt_einsum, six, torch, torchvision
 
 [tool:pytest]
 filterwarnings = error
     ignore::DeprecationWarning
-    once::DeprecationWarning
```
Comment (fritzo) [inline setup.cfg]: @fehiepsi Can you please revert this change? It is unrelated to the PR title and breaks previously tuned logging. These kind of tiny "cleanup" tweaks are not necessary. Let's spend time on useful things like GPyTorch integration.

Comment (fehiepsi) [reply setup.cfg]: @fritzo Though it might not be necessary, these works really help me relax my mind after reading hard-to-understand papers. I just do this clean-up job periodically for that reason.

Comment (neerajprad) [reply setup.cfg]: +1. Both `once` and `ignore` would be ignored by pytest. `once` will print the warning only for the first occurrence of such a warning, and `ignore` would not print the warning at all. The reason why it was set to `once` is so that we see warnings at the end of pytest summary about deprecated methods in external dependencies, and can take appropriate action ahead of time. Let us not remove this! :)

Comment (fehiepsi) [reply setup.cfg]: It seems complicated. Haizzz....

Comment (fritzo) [reply setup.cfg]: @fehiepsi understood, I also find it relaxing to clean up after reading papers. Just remember that cleaning up Pyro also incurs review overhead, and reviewing subtle config PRs may not be relaxing to reviewers :wink:
